README.md

```diff
@@ -0,0 +1,79 @@
+# Environment Variables Injector
+
+This extension is designed to help users export snap options configured with the [snap set command](https://snapcraft.io/docs/configuration-in-snaps) into environment variables.
+
+Additionally, this extension supports environment files, which we refer to as `envfile` for brevity.
+
+To prevent conflicts with other snap configurations and to maintain some level of isolation between apps, the following rules apply:
+
+* **Global:** Environment variables are visible to all apps within the snap that use this extension.
+* **Local:** Environment variables are visible only to the specific `app` defined during the snap set command.
+
+|        | Snap Configuration Option      | Environment Variable |
+|--------|--------------------------------|----------------------|
+| Global | env.<key>=<value>              | <KEY>=<value>        |
+| Local  | apps.<app>.env.<key>=<value>   | <KEY>=<value>        |
+
+**Examples**
+
+Assuming `my-snap` is the name of the snap that uses the env-injector extension:
+
+* `sudo snap set <my-snap> env.default_port=8080` will be mapped to `DEFAULT_PORT=8080` globally.
+
+* `sudo snap set <my-snap> apps.server.env.endpoint="/v2/store"` will be mapped to `ENDPOINT="/v2/store"` and will be visible only to the `server` app.
+
+For environment files, where `path` refers to the `envfile` path:
+
+|        | Snap Configuration Option    |
+|--------|------------------------------|
+| Global | envfile=<path>               |
+| Local  | apps.<app>.envfile=<path>    |
+
+**Examples**
+
+* `sudo snap set <my-snap> envfile=$HOME/envfile.env` makes all the environment variables globally visible.
+
+* `sudo snap set <my-snap> apps.server.envfile=$HOME/envfile.env` makes the environment variables visible only to the `server` app.
+
+## Syntax and Rules
+
+The precedence of environment variables follows this order:
+
+1. Env vars from local snap config options.
+2. Env vars from global snap config options.
+3. Env vars extracted from the **local** `envfile`.
+4. Env vars extracted from the **global** `envfile`.
```

Comment:
Reading this sounds like the global env file has the highest precedence, applied in the end.

Reply:
Actually, this is the order of precedence that we use, but we can also list by **the order which is processed/applied**, which will be exactly the inverse of this list.

Reply:
You could change the first sentence to "The environment variables are processed in the following order:" to avoid ambiguity.

Reply:
Done! Thank you!